### ISSUE TITLE
[Documentation] Diference between Doc and Package

### ISSUE THREAD

**Issue by allanbarbosa** [particle-php/Validator#180]
### What

Hello, good morning.
I am using your package and i notice that the documentation about the NotEmpty validate on optional fields is not equal in the package code. In your documentation you said that the optional validation accept by default the allowEmpty = false, but in the code package is inverse, the allowEmpty = true.

### What is expected

Change the code package or change the documentation.

Thank you so much
Sorry for my bad english, i'm trainning.

**Comment by RickvdStaaij** [comment]
@allanbarbosa good catch, in `optional($key, $name = null, $allowEmpty = true)` allow empty is true by default, instead of false as the documentation suggests.

@berry-langerak do you know if there was a specific reason for this? Or should I revert it to be false by default.

**Comment by allanbarbosa** [comment]
@RickvdStaaij thank you, this is a excelent package. Good job.

**Comment by berry-langerak** [comment]
Hi. If something is optional, I think the most logical default would be to allow the value to be empty. Right?

**Comment by RickvdStaaij** [comment]
Hmm I'm not sure if that's the most logical, I can't really decide. But I think that it's probably better to adjust the documentation at this point. It might lead to unexpected behavior for people who rely on this principle. I'll create a PR to update the docs soon. Agreed? :)
